Fix TORCH_FEATURE_VERSION guards

mikaylagawarecki opened the pull request:
This is tested by #167962 which ensures we get compilation errors when using functions that convert Device/HeaderOnlyArrayRef to StableIValue and target 2.9

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #168062
* #168034
* #167962
* #167804
* #167803
* __->__ #167802
* #168025

mikaylagawarecki committed:
Fix TORCH_FEATURE_VERSION guards

[ghstack-poisoned]

janeyx99 reviewed (commented):
stableivalue_conversions.h makes sense to me, but not ops.h

mikaylagawarecki committed:
Update on "Fix TORCH_FEATURE_VERSION guards"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Fix TORCH_FEATURE_VERSION guards"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Fix TORCH_FEATURE_VERSION guards"




[ghstack-poisoned]

janeyx99 commented on torch/csrc/stable/stableivalue_conversions.h:
```diff
@@ -126,6 +152,18 @@ struct FromImpl<ScalarType> {
+// This conversion is only used
+// (1) In the conversion for torch::stable::Device (already gated)
+// (2) Within the user extension to translate between libtorch/extension's
+//     DeviceType (no gating needed)
```
v good note thank u

janeyx99 reviewed (approved):
approving trusting that the prs stacked on top of this test this

mikaylagawarecki committed:
Update on "Fix TORCH_FEATURE_VERSION guards"

This is tested by #167962 which ensures we get compilation errors when using functions that convert Device/HeaderOnlyArrayRef to StableIValue and target 2.9




[ghstack-poisoned]
